tests/network/__init__.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright 2023 Google LLC.
+#  Copyright (c) Microsoft Corporation.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Literal
+
+from test_helpers import (execute_command, send_JSON_command, subscribe,
+                          wait_for_event)
+
+
+async def create_dummy_blocked_request(
+    websocket, context_id: str, *, url: str,
+    phases: list[Literal["beforeRequestSent", "responseStarted",
+                         "authRequired"]]):
+    """Creates a dummy blocked network request and returns its network id."""
+
+    await subscribe(websocket, ["cdp.Fetch.requestPaused"])
```

Comment:
Let's rely on BiDi whenever possible. In this case, we can rely on the `network.beforeRequestSent` or `network.responseStarted` BiDi events, depending on the `phases`.

Reply:
Note that I already left a TODO below to refactor this. I do agree, but will defer to another PR.